### ISSUE TITLE
no-invalid-this should not complain on functions with contextual this

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3022]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: gulp

#### TypeScript code being linted

```ts
export function f<T>(this: Observable<T>): Observable<T> {
 return this.map(i=>i); // this here is defined by parameter
}
```

with `tslint.json` configuration:

```json
"no-invalid-this": true
```

#### Actual behavior
Error

#### Expected behavior
No error

**Comment by ajafff** [comment]
Fixed by #3267
